### PR TITLE
Picking up former-missing build scripts

### PR DESCRIPTION
**Pull request by 6hundreds**
Fixes https://github.com/gradle/gradle/issues/18897

### CODE REVIEW

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        }
+
+        def b = createDir('b')
+
+        and:
+        configurationCacheRun 'ok'
+
+        and:
+        b.file('build.gradle') << """
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy:42]: 🧪 `build.gradle.kts` should be tested as well.

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            file('build.gradle') << """
+                task ok
+        """
+        }
+
+        and:
+        configurationCacheRun 'ok'
+
+        and:
+        settingsFile << """
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy:75]: 🧪 `settings.gradle.kts` should be tested as well.

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            file('build.gradle') << """
+                task bar
+        """
+        }
+
+        and:
+        configurationCacheRun 'ok'
+
+        and:
+        file("buildSrc/settings.gradle") << """
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy:107]: 🧪 `settings.gradle.kts` should be tested as well.

**Review by bamboo — CHANGES_REQUESTED** [review]

**Comment by 6hundreds** [comment]
@bot-gradle test ACC

**Comment by 6hundreds** [comment]
@bot-gradle test ACC

**Comment by 6hundreds** [comment]
@bot-gradle test ACC

File: subprojects/core-api/src/main/java/org/gradle/internal/scripts/DefaultScriptFileResolver.java [hunk 4]
```diff
@@ -48,13 +64,20 @@ public List<File> findScriptsIn(File dir) {
         }
         List<File> found = new ArrayList<File>(candidates.length);
         for (File candidate : candidates) {
-            if (candidate.isFile() && hasScriptExtension(candidate)) {
+            if (isCandidateFile(candidate) && hasScriptExtension(candidate)) {
```
Comment (6hundreds) [inline subprojects/core-api/src/main/java/org/gradle/internal/scripts/DefaultScriptFileResolver.java:67]: @bamboo Should we also notifying from this place?

Comment (bamboo) [reply subprojects/core-api/src/main/java/org/gradle/internal/scripts/DefaultScriptFileResolver.java:67]: Not really. Eventually we will need to instrument the `dir.listFiles()` call above (somehow only considering script extensions) but that's for later.

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by 6hundreds** [comment]
@bot-gradle test this

**Comment by 6hundreds** [comment]
@bot-gradle test this

**Comment by 6hundreds** [comment]
@bamboo Should we also add test coverage for included build cases?

**Comment by 6hundreds** [comment]
@bot-gradle test this

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        fixture.addMissingScript()
+
+        when:
+        configurationCacheRun 'ok'
+
+        then:
+        outputContains(fixture.expectedCacheInvalidationMessage)
+
+        where:
+        missingScriptsSpec << MissingScriptFixture.specs()
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMissingScriptIntegrationTest.groovy:44]: 👍

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractAutoTestedSamplesTest.groovy [hunk 6]
```diff
@@ -29,6 +31,9 @@ class AbstractAutoTestedSamplesTest extends AbstractIntegrationTest {
             if (tagSuffix.contains('WithDeprecations')) {
                 executer.noDeprecationChecks()
             }
+            if (tagSuffix.contains('WithoutCC') && GradleContextualExecuter.configCache) {
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractAutoTestedSamplesTest.groovy:34]: Would it make sense to use `Assume.assumeFalse(...)` here?

**Review by bamboo — APPROVED** [review]
Great job! 👍

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
@bot-gradle cancel

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractAutoTestedSamplesTest.groovy (hunk 6 above)
Comment (bamboo) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractAutoTestedSamplesTest.groovy]: It wouldn't.

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**